### PR TITLE
Make OpenTSDB metric prefix configurable.

### PR DESCRIPTION
**Pull request by dpoltavets**
- OpenTSDB metric prefix is configurable;
- `pen.` is replaced by `kilda.`.

Closes: #1630, #1953.

---
This change is [Reviewable](https://reviewable.io/reviews/telstra/open-kilda/2026)

### CODE REVIEW

**Comment by dpoltavets** [comment]
___
*[confd/templates/atdd-staging/kilda.properties.tmpl, line 44 at r1](https://reviewable.io/reviews/telstra/open-kilda/2026#-LYRfXEdBhFzjZN4HMRq:-LYVR7jV25QkwRakOX93:ba70oie) ([raw file](https://github.com/telstra/open-kilda/blob/4f7f1cc67a072656060118801a0b0a7c740cef5f/confd/templates/atdd-staging/kilda.properties.tmpl#L44)):*
<details><summary><i>Previously, pzakatov (Pavel Zakatov) wrote…</i></summary><blockquote>

I'd suggest making a variable name more specific to the opentsdb area, like `kilda_opentsdb_metric_prefix`. Thoughts?
</blockquote></details>

You are right. Fixed.

**Comment by dpoltavets** [comment]
___
*[confd/templates/wfm/topology.properties.tmpl, line 23 at r1](https://reviewable.io/reviews/telstra/open-kilda/2026#-LYRkO375KubrZAA9LFu:-LYVRCjg76I2SjRSc5V0:b-j7cxhv) ([raw file](https://github.com/telstra/open-kilda/blob/4f7f1cc67a072656060118801a0b0a7c740cef5f/confd/templates/wfm/topology.properties.tmpl#L23)):*
<details><summary><i>Previously, pzakatov (Pavel Zakatov) wrote…</i></summary><blockquote>

Should it be a part of `opentsdb.` group as well?
</blockquote></details>

Yes, it should be. Fixed.

**Comment by dpoltavets** [comment]
___
*[confd/vars/main.yaml, line 100 at r1](https://reviewable.io/reviews/telstra/open-kilda/2026#-LYRgH0O5NAXNDnsZosK:-LYVRRRY5IKvHD6kBBvU:bt2vr1b) ([raw file](https://github.com/telstra/open-kilda/blob/4f7f1cc67a072656060118801a0b0a7c740cef5f/confd/vars/main.yaml#L100)):*
<details><summary><i>Previously, pzakatov (Pavel Zakatov) wrote…</i></summary><blockquote>

There is supposed to be a dot at the end. like `kilda.`
</blockquote></details>

Fixed.

**Comment by dpoltavets** [comment]
___
*[services/wfm/src/main/java/org/openkilda/wfm/share/utils/MetricFormatter.java, line 28 at r1](https://reviewable.io/reviews/telstra/open-kilda/2026#-LYRhJp5Bc_8GHXjLjFA:-LYVRVrP6v0JiY8SAHy6:b-1o4vaq) ([raw file](https://github.com/telstra/open-kilda/blob/4f7f1cc67a072656060118801a0b0a7c740cef5f/services/wfm/src/main/java/org/openkilda/wfm/share/utils/MetricFormatter.java#L28)):*
<details><summary><i>Previously, pzakatov (Pavel Zakatov) wrote…</i></summary><blockquote>

Have you considered a case when there is no prefix configured? I'd propose to have a dot as part of the prefix.
</blockquote></details>

I agree. Fixed.

**Comment by dpoltavets** [comment]
___
*[services/wfm/src/main/resources/topology.properties.example, line 50 at r2](https://reviewable.io/reviews/telstra/open-kilda/2026#-LYiO2Jw7B0BNAigRQ5C:-LYjnQ9UAcxS0nkwansb:b-ffh723) ([raw file](https://github.com/telstra/open-kilda/blob/6a4ace3c66e28da84dc22849c53f40402517f717/services/wfm/src/main/resources/topology.properties.example#L50)):*
<details><summary><i>Previously, pzakatov (Pavel Zakatov) wrote…</i></summary><blockquote>

Minor, should it be "kilda." (with a dot at the end) here? 
</blockquote></details>

Yes, it should be. Thanks! Fixed.

File: services/wfm/src/main/java/org/openkilda/wfm/topology/islstats/IslStatsTopology.java [hunk 1]
```diff
@@ -40,15 +40,15 @@ public StormTopology createTopology() {
...
+        builder.setSpout(ISL_STATS_SPOUT_ID, buildKafkaSpout(topoDiscoTopic, ISL_STATS_SPOUT_ID));
 
-        IslStatsBolt verifyIslStatsBolt = new IslStatsBolt();
+        IslStatsBolt verifyIslStatsBolt = new IslStatsBolt(topologyConfig.getMetricPrefix());
         logger.debug("starting {} bolt", ISL_STATS_BOLT_ID);
         builder.setBolt(ISL_STATS_BOLT_ID, verifyIslStatsBolt, topologyConfig.getParallelism())
                 .shuffleGrouping(ISL_STATS_SPOUT_ID);
 
         String openTsdbTopic = topologyConfig.getKafkaOtsdbTopic();
-        KafkaBolt openTsdbBolt = createKafkaBolt(openTsdbTopic);
```
Comment (niksv) [inline services/wfm/src/main/java/org/openkilda/wfm/topology/islstats/IslStatsTopology.java]: please do not change `createKafkaBolt` and `createKafkaSpout` yet. It causes `Could not deserialize message=00:00:00:00:00:00:00:07`

we need to fix #1985 first

File: services/wfm/src/main/java/org/openkilda/wfm/topology/islstats/IslStatsTopologyConfig.java [hunk 2]
```diff
@@ -29,4 +30,7 @@ default String getKafkaTopoDiscoTopic() {
     default String getKafkaOtsdbTopic() {
         return getKafkaTopics().getOtsdbTopic();
     }
+
+    @Key("opentsdb.metric.prefix")
```
[the remark this answers is not in the document]
Comment (dpoltavets) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/islstats/IslStatsTopologyConfig.java]: Added.

File: services/wfm/src/main/java/org/openkilda/wfm/topology/ping/PingTopologyConfig.java [hunk 3]
```diff
@@ -67,6 +67,9 @@ default String getKafkaOtsdbTopic() {
         return getKafkaTopics().getOtsdbTopic();
     }
 
+    @Key("opentsdb.metric.prefix")
```
[the remark this answers is not in the document]
Comment (dpoltavets) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/ping/PingTopologyConfig.java]: Added.

File: services/wfm/src/main/java/org/openkilda/wfm/topology/portstate/PortStateTopologyConfig.java [hunk 4]
```diff
@@ -37,4 +38,7 @@ default String getKafkaStatsTopic() {
     default String getKafkaSpeakerTopic() {
         return getKafkaTopics().getSpeakerTopic();
     }
+
+    @Key("opentsdb.metric.prefix")
```
[the remark this answers is not in the document]
Comment (dpoltavets) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/portstate/PortStateTopologyConfig.java]: Added.

File: services/wfm/src/main/java/org/openkilda/wfm/topology/stats/StatsTopologyConfig.java [hunk 5]
```diff
@@ -33,4 +34,7 @@ default String getKafkaStatsTopic() {
     default String getKafkaSpeakerFlowTopic() {
         return getKafkaTopics().getSpeakerFlowTopic();
     }
+
+    @Key("opentsdb.metric.prefix")
```
[the remark this answers is not in the document]
Comment (dpoltavets) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/stats/StatsTopologyConfig.java]: Added.

File: services/wfm/src/main/java/org/openkilda/wfm/topology/islstats/IslStatsTopology.java (hunk 1 above)
Comment (dpoltavets) [reply services/wfm/src/main/java/org/openkilda/wfm/topology/islstats/IslStatsTopology.java]: Fixed.

**Pull request merged**